### DISCUSSION TITLE
ESP32 as sensor?

### DISCUSSION THREAD

**Discussion by bennop** [mKeRix/room-assistant#731] [Ideas]
Just found out about this exciting project ...

While going through the pages I was wondering whether it might be possible to use ESP32s as sensors?

I have already deployed some via esphome and as they support both bluetooth and wifi, they might work as an even smaller (and cheaper) sensor, possibly "reporting" to raspberry-based room assistant instance ...

While I lack the technical knowledge to judge the feasibility  let alone implementing this I wanted to at least raise the point.

**Comment by smmason** [comment]
Seconded!  I have quite a few ESP32s in use for door/window/motion/temp sensing. Would love to add them to Room Assistant.

**Reply by MvQuast** [reply]
like the idea - and even smaller than Pi Zero

**Comment by dhallgb** [comment]
I'd love the idea, but at the moment they don't run a Linux stack - just an embedded language like MicroPython or Processing (which is similar to C/C++ via the Arduino IDE). room-assistant needs other JavaScript libraries and underlying OS-levels calls, so it would take a complete re-write to run in on embedded hardware like the ESP32. However, ESP8266 and ESP32 programming is relatively simple and given enough motivation and time it could be possible - also see #762 for someone who may already be doing this.
